app/api/imports/shapefile.py

```diff
@@ -0,0 +1,44 @@
+import os
+import tempfile
+import zipfile
+from django.contrib.gis.geos import MultiPolygon, Polygon
+
+
+def extract_zip_to_temp_dir(filename):
+    """Extracts zipfile into a new temporary folder."""
+    temp_dir = tempfile.mkdtemp()
+    unzipper = zipfile.ZipFile(filename)
+    unzipper.extractall(temp_dir)
```

Comment:
Does this handle the weird OS X zipfiles issue? The HunchLab importer has this line: `files_to_extract = [f for f in unzipper.namelist() if '__MACOSX' not in f]`